controllers/rack.go

```diff
@@ -702,31 +685,46 @@ func (r *SingleClusterReconciler) rollingRestartRack(
 			" config",
 	)
 
-	for _, pod := range podList {
+	// Find pods which needs restart
+	var podsToRestart []*corev1.Pod
+	for i := range podList {
+		pod := podList[i]
+
 		// Check if this pod needs restart
 		restartType, err := r.getRollingRestartTypePod(rackState, pod)
 		if err != nil {
 			return found, reconcileError(err)
 		}
 
 		if restartType == NoRestart {
-			r.Log.Info(
-				"This Pod doesn't need rolling restart, Skip this", "pod",
-				pod.Name,
-			)
+			r.Log.Info("This Pod doesn't need rolling restart, Skip this", "pod", pod.Name)
 			continue
 		}
+		podsToRestart = append(podsToRestart, &pod)
+	}
 
-		res := r.rollingRestartPod(rackState, pod, restartType, ignorablePods)
+	// Failed pods should always come before active pods.
```

Comment:
Can we batch failed pods and active pods separately and not mix them in single batch? doing this we can make sure failed pods batch gets handled first.